### PR TITLE
Does not allow updating an existing allowance to a non-zero value

### PR DESCRIPTION
**Pull request by pdyraga**
Does not allow updating an existing allowance to a value that is non-zero to
avoid someone using both the old and the new allowance by unfortunate
transaction ordering. To update an allowance to a non-zero value one should
update to zero first or use `increaseBalanceAllowance` or
`decreaseBalanceAllowance` for an atomic update. We have been warning about this
case in `approveBalance` documentation, but we were not enforcing this behavior.
Now we have a `require` making sure the rule is obeyed.

### CODE REVIEW

File: solidity/test/bank/Bank.test.ts [hunk 1]
```diff
@@ -424,15 +424,33 @@ describe("Bank", () => {
...
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should replace the previous allowance with zero", async () => {
+          expect(await bank.allowance(owner.address, spender)).to.equal(0)
+        })
+      })
+
+      context("when trying to overwrite with a non-zero value", () => {
```
Comment (nkuba) [inline solidity/test/bank/Bank.test.ts:448]: Can we add a test case where the current approval is zero and the new approval is zero?

Comment (pdyraga) [reply solidity/test/bank/Bank.test.ts:448]: Sure, on it.

Comment (pdyraga) [reply solidity/test/bank/Bank.test.ts:475]: d782d5800b4da239d1346b33e6ae0c4d89cbfa5c

**Review by nkuba — APPROVED** [review]
